### PR TITLE
Go back to urfave v1

### PR DESCRIPTION
**Pull request by ibuildthecloud**
github.com/rancher/spur is not a project we would like to maintain going forward so this PR goes back to urfave v1 but does config file support using a very naive approach of pre-processing and manipulating os.Args before they are passed to urfave. The actual changes to add the config file support are trivial but this PR is much larger because it reverts all the spur changes. Features that could see a regression would be the --debug, --enable-selinux, --protect-kernel-defaults flags as those changes were entangled in the spur changes.

### CODE REVIEW

File: pkg/cli/cmds/agent.go [hunk 1]
```diff
@@ -155,7 +153,7 @@ var (
 		Usage:       "(agent/node) Enable SELinux in containerd",
```
Comment (dweomer) [inline pkg/cli/cmds/agent.go:153]: copy pasta on usage for `protect-kernel-defaults`

File: pkg/cli/cmds/config.go [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-		Aliases: []string{"c"},
-		Usage:   "(config) Load configuration from `FILE`",
-		EnvVars: []string{"K3S_CONFIG_FILE"},
-		Value:   DefaultConfig,
+	// ConfigFlag is here to show to the user, but the actually processing is done by configfileargs before
+	// call urfave
+	ConfigFlag = cli.StringFlag{
+		Name:   "config,c",
+		Usage:  "(config) Load configuration from `FILE`",
+		EnvVar: "K3S_CONFIG_FILE",
```
Comment (dweomer) [inline pkg/cli/cmds/config.go:14]:
```suggestion
		EnvVar: version.ProgramUpper + "_CONFIG_FILE",
```

File: pkg/cli/server/server.go [hunk 3]
```diff
@@ -263,7 +266,7 @@ func run(app *cli.Context, cfg *cmds.Server) error {
 	}
 
 	agentConfig := cmds.AgentConfig
-	agentConfig.Debug = app.Bool("debug")
+	agentConfig.Debug = app.GlobalBool("bool")
```
Comment (dweomer) [inline pkg/cli/server/server.go:269]:
```suggestion
	agentConfig.Debug = app.GlobalBool("debug")
```

File: pkg/cli/cmds/agent.go (hunk 1 above)
Comment (dweomer) [reply pkg/cli/cmds/agent.go:153]: This looks like a new flag, what is it trying to do?

**Comment by cjellick** [comment]
https://github.com/rancher/k3s/issues/2228

**Pull request merged**